entry/src/block_component.rs

```diff
@@ -415,6 +438,23 @@ impl BlockComponent {
         matches!(self, Self::BlockMarker(_))
     }
 
+    /// Fuses another BlockComponent into this one if both are entry batches.
+    ///
+    /// Returns None if fusion occurred, Some(other) if no fusion was possible.
+    pub fn maybe_fuse(&mut self, other: Self) -> Option<Self> {
+        match self {
+            Self::EntryBatch(entries) => {
+                if let Self::EntryBatch(other_entries) = other {
+                    entries.extend(other_entries);
+                    None
+                } else {
+                    Some(other)
+                }
+            }
+            _ => Some(other),
+        }
```

Comment:
```suggestion
        match (self, &other) {
            (Self::EntryBatch(self_entries), Self::EntryBatch(other_entries)) => {
                self_entries.extend_from_slice(other_entries);
                None
            }
            _ => Some(other),
        }
    }
```

I find the suggestion to be shorter, more readable, and it uses extend_from_slice which can be faster in many cases.

Reply:
done